core/templates/dev/head/services/SearchService.ts

```diff
@@ -183,7 +183,9 @@ angular.module('oppia').factory('SearchService', [
             updateSearchFields(itemsType, urlComponent, selectionDetails);
           } catch (error) {
             selectionDetails[itemsType].selections = {};
-            throw error;
+            if (itemsType !== 'languageCodes') {
```

Comment:
I think this approach is too broad-brush and doesn't really get at the core issue. The fix I would expect is to examine the urlComponent, "clean" it, update the URL accordingly, and then execute the query.

Reply:
I've changed this, now it cleans the urlComponents (ie. removes them form the URL) if they are malformed.

Reply:
It doesn't seem to update the URL in the address bar? (See my previous comment.)

Reply:
Setting `selectionDetails[itemsType].selections = {};` in the step above takes care of updating the URL. It sets the `ctrl.selectionDetails.<itemType>.selection` to empty, this is used later in `_getSuffixForQuery` method in SearchService where the appropriate suffix (based on available selections).

I've tested the different ways `_getSuffixForQuery` can be called and have verified that the URL updates correctly.